src/main/java/io/anserini/collection/TrecwebCollection.java

```diff
@@ -19,22 +19,37 @@
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 /**
  * Class representing an instance of a TREC web collection.
  */
-public class TrecwebCollection extends TrecCollection  {
+public class TrecwebCollection extends DocumentCollection
+    implements FileSegmentProvider<TrecwebCollection.Document> {
 
-  public class FileSegment extends TrecCollection.FileSegment {
-    public FileSegment(Path path) throws IOException {
-      super(path);
-      dType = new Document();
-    }
+  @Override
+  public FileSegment<TrecwebCollection.Document> createFileSegment(Path p) throws IOException {
+    return new FileSegment(p);
   }
 
   @Override
-  public FileSegment createFileSegment(Path p) throws IOException {
-    return new FileSegment(p);
+  public List<Path> getFileSegmentPaths() {
+    Set<String> skippedFilePrefix = new HashSet<>(Arrays.asList("readme"));
```

Comment:
It seems that wt10g and gov2 are pretty clean on tuna - there are no `readme` file or other unwanted folders. So we probably do not need these filters?